var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -125,3 +125,22 @@ def install(self, spec, prefix):
             ]
 
         make('install', *make_args)
+
+    @property
+    def libs(self):
+        """Export the libraries of SuiteSparse.
+        Sample usage: spec['suite-sparse'].libs.ld_flags
+                      spec['suite-sparse:klu,btf'].libs.ld_flags
+        """
+        # Component libraries, ordered by dependency. Any missing components?
+        all_comps = ['klu', 'btf', 'umfpack', 'cholmod', 'colamd', 'amd',
+                     'camd', 'ccolamd', 'cxsparse', 'ldl', 'rbio', 'spqr',
+                     'suitesparseconfig']
+        query_parameters = self.spec.last_query.extra_parameters
+        comps = all_comps if not query_parameters else query_parameters
+        libs = find_libraries(['lib' + c for c in comps], root=self.prefix.lib,
+                              shared=True, recursive=False)
+        if not libs:
+            return None
```

Comment:
I missed this subtle point on first review. Why are you returning `None` here? It's not clear to me how this should be handled. If clients are supposed to check for it (C-style):
```python
libs = spec['suite-sparse'].libs
if libs is None:
    # Do something
    ...
```
probably something like:
```python
libs = find_libraries(['lib' + c for c in comps], root=self.prefix.lib, shared=True, recursive=False)
if not libs:
    raise LibrariesNotFoundError(msg)

libs += find_system_libraries('librt')
return libs
```
would be more appropriate, in case we want to be defensive? Otherwise, if we assume we can return an empty list we can do:
```python
libs = find_libraries(['lib' + c for c in comps], root=self.prefix.lib, shared=True, recursive=False)
if libs:
    libs += find_system_libraries('librt')
return libs
```
What do you think? 

p.s. other than this small doubt, I think this PR is ready to be merged

Reply:
I think raising an error is the appropriate behavior.

Right now, returning `None` from `.libs` raises an AttributeError which initially seemed like an easy way to report an error. However, the generated message does not quite fit this usage and is, therefore, somewhat misleading - it says that the package has no such attribute. That's why, I proposed a modification in PR #7277  that reports such cases as query failures and raises RuntimeError.

That said, I'm open to any other solution.

Reply:
> raise LibrariesNotFoundError(msg)

👍

Reply:
Since #7277 was merged, returning `None` will raise an error with appropriate message.

Is there anything else stopping this from being approved and merged?